carrier.go

```diff
@@ -50,34 +48,59 @@ type CarrierType struct {
 }
 
 type carrierAccountRequest struct {
-	CarrierAccount *CarrierAccount `json:"carrier_account,omitempty"`
+	Data *CarrierAccount `json:"carrier_account,omitempty"`
 }
 
-func (c *Client) selectCarrierAccountCreationEndpoint(account CarrierAccount) string {
+// UpsCarrierAccountCreationParameters contains the parameters needed to create a new UPS carrier account.
+type UpsCarrierAccountCreationParameters struct {
+	Type          string `json:"type,omitempty"`
+	Description   string `json:"description,omitempty"`
+	Reference     string `json:"reference,omitempty"`
+	AccountNumber string `json:"account_number,omitempty"`
+}
+
+type upsCarrierAccountCreationRequest struct {
+	Data *UpsCarrierAccountCreationParameters `json:"ups_oauth_registrations,omitempty"`
+}
+
+// UpsCarrierAccountUpdateParameters contains the parameters needed to update a UPS carrier account.
+type UpsCarrierAccountUpdateParameters struct {
+	// TODO: Add Description and Reference if/when API allows updating them.
```

Comment:
suggestion: let's remove this TODO as it's uncertain if/when this will happen and instead create an internal ticket to address this holistically across all the libs. Then we can carry on the convo with the team that is building the API to determine if/when it can go down.